### PR TITLE
new status summary widget, plus lots of fixes

### PR DESCRIPTION
**Pull request by indirectlylit**
### Summary

* new wrapper components for displaying consistent status summary views
* state transitions
* many small styling improvements
* upgraded a few instances to new component
* added a test page for viewing the new widget states at [http://localhost:8000/coach/#/about/statuses](http://localhost:8000/coach/#/about/statuses)
* adds a new prototype 'help needed' indicator in progress bars, also disabled by default

### Reviewer guidance

~NOTE - this PR depends on #4844, which should be reviewed and merged first~

### References

…

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Gherkin stories have been updated
- [ ] Unit tests have been updated

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by rtibbles** [comment]
Have restarted failed build, as it appeared to be a random error.

File: kolibri/plugins/coach/assets/src/views/common/status/HelpNeeded.vue [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  import LearnerProgressCount from './LearnerProgressCount';
+  import LearnerProgressRatio from './LearnerProgressRatio';
+
+  export default {
+    name: 'HelpNeeded',
+    components: {
+      // eslint-disable-next-line vue/no-unused-components
+      LearnerProgressCount, // it is used, it's just referenced dynamically
+      // eslint-disable-next-line vue/no-unused-components
+      LearnerProgressRatio, // it is used, it's just referenced dynamically
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/common/status/HelpNeeded.vue:29]: @jonboiser @rtibbles have either of you encountered this before? I'm surprised the linter is complaining

**Comment by indirectlylit** [comment]
should be ready for review & merge. when it's in I'll rebase https://github.com/learningequality/kolibri/pull/4861

File: kolibri/plugins/coach/assets/src/views/common/status/HelpNeeded.vue (hunk 1 above)
Comment (jonboiser) [reply kolibri/plugins/coach/assets/src/views/common/status/HelpNeeded.vue:29]: its a bug. there's a patch on the eslint library to fix it.

**Comment by jonboiser** [comment]
On QuizzesBloc, the 'needs help' icon should not appear at all. On LessonsBlock, it should not appear if there are 0 learners needing help

**Review by jonboiser — APPROVED** [review]

**Comment by indirectlylit** [comment]
> On QuizzesBloc, the 'needs help' icon should not appear at all. On LessonsBlock, it should not appear if there are 0 learners needing help

change pushed

**Pull request merged**